copy/copy.go

```diff
@@ -1166,6 +1166,18 @@ func computeDiffID(stream io.Reader, decompressor compression.DecompressorFunc)
 	return digest.Canonical.FromReader(stream)
 }
 
+type errorAnnotationReader struct {
+	reader io.Reader
+}
+
+func (r errorAnnotationReader) Read(b []byte) (n int, err error) {
+	n, err = r.reader.Read(b)
+	if err != nil && err != io.EOF {
+		return n, errors.Wrapf(err, "error happened during read")
```

Comment:
Absolutely non-blocking: Maybe just `err = errors.Wrapf(…)` and let it fall through here